### PR TITLE
Correct span for pub_restricted field

### LINKED ISSUE #35435: pub_restricted field warning spans include close-paren
Warnings on pub_restricted fields produce odd spans including the close-paren:

###### `src/lib.rs`

```rust
#![feature(pub_restricted)]
pub mod foo;
```

###### `src/foo.rs`

```rust
pub struct Foo {
    pub(super) bar: u8,
}
```

###### `cargo build`

```
warning: struct field is never used: `bar`, #[warn(dead_code)] on by default
 --> src/foo.rs:2:14
  |
2 |     pub(super) bar: u8,
  |              ^^^^^^^^^
```

###### `rustc --version`

```
rustc 1.12.0-nightly (b30eff7ba 2016-08-05)
```

### PR DESCRIPTION
**Pull request by sanxiyn**
Fix #35435.

### CODE REVIEW

File: src/test/compile-fail/pub-struct-field-span.rs [hunk 1]
```diff
@@ -18,6 +20,11 @@ struct Foo {
     //~^ error: field `bar` is already declared [E0124]
     bar: u8,
 
+    pub
```
Comment (nikomatsakis) [inline src/test/compile-fail/pub-struct-field-span.rs]: May I suggest you use a `ui` test? That way you can actually directly record the highlight that is expected, giving more precision.

**Comment by sanxiyn** [comment]
Changed test to ui test.

**Pull request merged**